openflow15/group.go

```diff
@@ -433,3 +433,110 @@ func (prop *GroupBucketPropWatch) UnmarshalBinary(data []byte) (err error) {
 
 	return
 }
+
+type NTRSelectionMethodType string
+
+const (
+	OFPGPT_EXPERIMENTER   = 0xFFFF
+	NTR_VENDOR_ID         = 0x0000154d
+	NTRT_SELECTION_METHOD = 1
+
+	NTR_HASH    NTRSelectionMethodType = "hash"
+	NTR_DP_HASH NTRSelectionMethodType = "dp_hash"
+)
+
+type NTRSelectionMethod struct {
+	Type             uint16
+	Length           uint16
+	ExperimenterID   uint32
+	ExperimenterType uint32
+	// padding with 4 bytes.
+	SelectionMethod [16]byte
+	SelectionParam  uint64
+	// Note, a valid field is supposed to configure with "HasMask==false", and its Value is to apply
```

Comment:
what does it mean "HasMask==false"? We can't use masked field here?

Reply:
It means when we construct a MatchField object, the "HasMask" property must be set as false. So we only set "MatchField.Value", but not set "MatchField.Mask". The specified value is actually used to mask on the the given field in a packet.